src/lib/storage/table.cpp

```diff
@@ -85,4 +86,32 @@ std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) { return _chunks.at(ch
 
 std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
+void Table::compress_chunk(const ChunkID chunk_id) {
+  const auto input_chunk = get_chunk(chunk_id);
+  const auto column_count = input_chunk->column_count();
+  std::vector<std::thread> threads;
```

Comment:
you could use auto-to-stick here

```suggestion
  auto threads = std::vector<std::thread>();
```